lib/travis/model/job/queue.rb

```diff
@@ -30,6 +32,25 @@ def queues
       def default
         @default ||= new(Travis.config.default_queue)
       end
+
+      def sudo_detected?(config)
+        config.values_at(*custom_stages).compact.map do |value|
+          Array(value).reject { |s| s =~ /\s*#.*/ }.map { |s| s =~ /\b(sudo|ping)\b/ }.any?
+        end.any?
+      end
+
+      def custom_stages
+        @custom_stages ||= %w(
+          before_install
+          install
+          before_script
+          script
+          before_cache
+          after_success
+          after_failure
+          after_script
+        ).map(&:to_sym)
+      end
```

Comment:
I don't think them getting out of sync is a huge issue. Mainly I think this would mean that we add stages in travis-build that are missing here, which means that someone might use `sudo` and we don't detect it, and they get routed to the Docker queue.

Already the way we detect if someone uses sudo isn't completely accurated (for example, someone might use `sudo` in a script that is called from the .travis.yml, we don't detect that), and I think this way of doing it is good enough™.